tests/cbmc/proofs/s2n_array_free_p/s2n_array_free_p_harness.c

```diff
@@ -21,12 +21,25 @@ void s2n_array_free_p_harness()
 {
     /* Non-deterministic inputs. */
     struct s2n_array *array = cbmc_allocate_s2n_array();
-    __CPROVER_assume(s2n_result_is_ok(s2n_array_validate(array)));
 
+    /* Assumptions. */
     nondet_s2n_mem_init();
+    __CPROVER_assume(s2n_result_is_ok(s2n_array_validate(array)));
 
     /* Operation under verification. */
-    if(s2n_result_is_ok(s2n_array_free_p(&array))) {
+    s2n_result result = s2n_array_free_p(&array);
+    if (s2n_result_is_ok(result)) {
         assert(array == NULL);
+    } else {
+        assert(s2n_errno != S2N_ERR_FREE_STATIC_BLOB);
     }
+
+    /* Cleanup after expected error cases, for memory leak check. */
+    if (s2n_result_is_error(result)) {
```

Comment:
You can merge these conditions together.

Reply:
Merged as suggested. Thanks!